### PR TITLE
Automate Staking and association/disassociation tests 

### LINKED ISSUE #1205: Disassociation tests for TFE

### PR DESCRIPTION
**Pull request by Ditmir-Vega**
closes #1205 #1174 #1234 #1238 #1239

### CODE REVIEW

File: automation/features/step-definitions/staking.steps.js [hunk 1]
```diff
@@ -1,18 +1,66 @@
...
+    $('[data-testid="associate-radio-wallet"]').click({force:true})
+    browser.pause(1000)
+    browser.getByTestId('token-amount-input').setValue(0.00001)
+    browser.pause(1000)
+    browser.getByTestId('token-input-submit-button').click()
+    browser.waitUntil(() => browser.getWindowHandles().length > 1);
+    browser.switchWindow("MetaMask Notification");
+    $('button=Confirm').click()
+    browser.switchWindow("VEGA");
+    browser.pause(2000)
```
Comment (jtsang586) [inline automation/features/step-definitions/staking.steps.js:55]: There's a lot of test ids here that should be added to the page object. There's also a lot of implicit waits here, is it possible to do something like wait for element to be displayed?

File: automation/features/step-definitions/staking.steps.js [hunk 2]
```diff
@@ -47,5 +41,93 @@ When(/^I can see the maximum amount of tokens in my wallet are in the token inpu
...
+
+Then(/^the epoch countdown timer is counting down$/,()=>{
+const currentCountdownTimerText = browser.getByTestId('current-epoch-ends-in').getText()
+browser.pause(2100) // let some time pass 
+expect(browser.getByTestId('current-epoch-ends-in').getText()).not.toEqual(currentCountdownTimerText)
+})
+
+Then(/^I pause some seconds$/,()=>{
+browser.pause(600000)
+})
```
Comment (jtsang586) [inline automation/features/step-definitions/staking.steps.js:58]: This looks like an unused step

File: automation/features/step-definitions/staking.steps.js [hunk 3]
```diff
@@ -47,5 +41,93 @@ When(/^I can see the maximum amount of tokens in my wallet are in the token inpu
...
+When(/^I click the pending transactions button$/, () => {
+    browser.getByTestId('pending-transactions-btn').click()
+});
+
+Then(/^I can see the pending transactions modal is shown$/, () => {
+ browser.getByTestId('pending-transactions-modal').waitForDisplayed({timeout:20000,timeoutMsg:"pending transactions modal did not display"})
+expect(browser.getByTestId('etherscan-link')).toBeDisplayed()
+expect(browser.getByTestId('etherscan-link')).toHaveAttr('href')
+expect($('.transactions-modal__status')).toHaveText('Pending')
+});
```
Comment (jtsang586) [inline automation/features/step-definitions/staking.steps.js:132]: Maybe also put these put these into page objects?

**Review by jtsang586 — APPROVED** [review]

File: automation/features/step-definitions/staking.steps.js (hunk 2 above)
Comment (Ditmir-Vega) [reply automation/features/step-definitions/staking.steps.js:58]: it is used for debugging , i put it in between steps some times when i need to pause the test . will delete and push

**Review by jtsang586 — APPROVED** [review]

**Review by bglownia — COMMENTED** [review]
Looks fine

**Pull request merged**